crates/pico_macros/src/db.rs

```diff
@@ -0,0 +1,70 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+pub fn derive_db(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let struct_name = input.ident;
+
+    let output = quote! {
+        impl ::pico_core::database::Database for #struct_name {
+            fn storage(&self) -> &impl ::pico_core::storage::Storage<Self> {
+                &self.storage
+            }
+
+            fn storage_mut(&mut self) -> &mut impl ::pico_core::storage::StorageMut<Self> {
+                &mut self.storage
+            }
+
+            fn current_epoch(&self) -> ::pico_core::epoch::Epoch {
+                use ::pico_core::storage::Storage;
+                self.storage().current_epoch()
+            }
+
+            fn increment_epoch(&mut self) -> ::pico_core::epoch::Epoch {
+                use ::pico_core::storage::StorageMut;
+                self.storage_mut().increment_epoch()
+            }
+
+            fn get<T: Clone + 'static>(&mut self, id: ::pico_core::source::SourceId<T>) -> T {
+                use ::pico_core::{storage::Storage, container::Container};
+                let time_calculated = self.storage()
+                    .sources()
+                    .get(&id.key)
+                    .expect("node should exist. This is indicative of a bug in Isograph.")
```

Comment:
if we expect others to use pico (we should! they should!), maybe we should say `bug in Pico`